ee/control/control.go

```diff
@@ -0,0 +1,142 @@
+package control
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// ControlService is the main object that manages the control service. It is responsible for fetching
+// and caching control data, and updating consumers and subscribers.
+type ControlService struct {
+	logger          log.Logger
+	cancel          context.CancelFunc
+	requestInterval time.Duration
+	consumers       map[string]consumer
+	subscribers     map[string][]subscriber
+	data            dataProvider
```

Comment:
```suggestion
	dataGetter            dataProvider
```

This doesn't hold data, it just gets it, right? Feels like we should reflect that in the name. Counterpoint, `dataGetter.Get` feels redundant. Maybe `fetcher` instead? `fetcher.Get` doesn't sound bad